docs/prometheus-integration.md

```diff
@@ -106,3 +106,163 @@ The configuration file above can be used to deploy Prometheus Server with
 scraping configuration for Antrea services.
 To deploy this configuration use
 `kubectl apply -f build/yamls/antrea-prometheus.yml`
+
+# Antrea Prometheus Metrics
+Antrea Controller and Agents expose various metrics, some are provided by the 
+Antrea components and others are provided by 3rd party components which are used
+by the Antrea components.
+
+Below is a list of metrics, provided by the components and by 3rd parties.
+
```

Comment:
Hi Kobi, These help strings are visible through API right? Are these not visible from Prometheus server/Grafana? In addition, some of them are standard metrics and not Antrea specific. 
Overall question: Is it useful to add these in Antrea documentation?

Reply:
@srikartati Hi,
You're correct: you could obtain that info via API and they're visible via Prometheus/Grafana.
However, from a user/admin perspective, there's no difference between Antrea and 3rd party metrics: they're all used for monitoring the Antrea solution. When a certain metric is required for monitoring nobody cares if it was provided by the Antrea codebase, Go, or something else. And it's much easier to look these up within a doc than browsing through listboxes in Prometheus.
Listing metrics within a document is very common, and as there were multiple questions regarding supported metrics, I think that it's a good idea to have such doc.

Reply:
@ksamoray I see a couple of issues:
1. Every time a metric is added. This has to be continuously extended. This document may blow up in content and size. 
2. I feel most of the metrics "help" field is almost similar to what name is offering and not offering anything beyond.

I think the questions on slack came up because of missing network policy documentation, which will be added in the near future.
IMO, a statement about looking for "help" field through API/listboxes in Prometheus is probably good enough. I will leave it you and others to decide.

Reply:
Hi @srikartati,
Yeah that is correct. Documentation has to be maintained - and maybe organized differently as [here](https://docs.cilium.io/en/stable/configuration/metrics/) or [here](https://docs.projectcalico.org/reference/felix/prometheus) but that doesn't make it less needed.
About (2): that only implies that our "help" requires more attention :)
Can you imagine,  when we will hopefully have a whole lot of metrics, that a user will have to scroll a listbox through the help text of hundreds of metrics, just to find how the CPU utilization metric is called? That is not useful at all IMO.
I think that the fact that every product in the market offers such doc speaks for itself...

Reply:
Thanks for the links. They may also have the additional purpose of showcasing the metrics they are offering :)
Do you think putting metrics in tables is a better representation?
I haven't tried this representation myself but the example instruction seems not so difficult: https://help.github.com/en/github/writing-on-github/organizing-information-with-tables

Reply:
@srikartati tried a table... tables look awkward in a markdown doc due to the 80 char limit. 
Maybe if we ever move this info into the website :)

Reply:
Weird that multiple lines in the description column will look awkward. Table format would have been nice! 
NIT: I looked at the rich diff. Bullet points can be removed as metrics are anyway in bold.

Reply:
Some of the metrics names are long, e.g antrea_controller_network_policy_sync_duration_milliseconds
which doesn't leave much room for description under 80 char limit...
I'll remove the bullets.